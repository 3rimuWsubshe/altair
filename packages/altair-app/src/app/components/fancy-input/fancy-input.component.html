<div class="fancy-input-container">
  <input
    #fancyInputEl
    type="text"
    class="fancy-input-content fancy-input-element"
    [placeholder]="placeholder"
    [(ngModel)]="value"
    (input)="handleInput()"
    (scroll)="handleScroll()"
    (keydown)="handleKeydown()"
    (blur)="handleBlur()"
    (keydown.enter)="handleSubmit()"
  >
  <div class="fancy-input-backdrop">
    <div class="fancy-input-content fancy-input-highlights" #fancyInputHighlightsEl>
      <ng-container *ngFor="let section of highlightData.sections; trackBy:trackByIndex;">
        <span
          *ngIf="!section.type"
        >{{ section.content }}</span>
        <mark
          *ngIf="section.type === 'mark'"
          nz-tooltip
          nzTooltipTitle="Variable"
        >{{ section.content }}</mark>
      </ng-container>
    </div>
  </div>
</div>
