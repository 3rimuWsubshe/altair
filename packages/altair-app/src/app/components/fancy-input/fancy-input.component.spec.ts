import { expect, it, beforeEach, describe } from '@jest/globals';
import { async, ComponentFixture, TestBed, tick, fakeAsync } from '@angular/core/testing';

import { FancyInputComponent } from './fancy-input.component';
import { FormsModule } from '@angular/forms';

describe('FancyInputComponent', () => {
  let component: FancyInputComponent;
  let fixture: ComponentFixture<FancyInputComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ FancyInputComponent ],
      imports: [
        FormsModule,
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(FancyInputComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should set component value when input value changes', async() => {
    const input = fixture.nativeElement.querySelector('input');
    input.value = 'some text';
    input.dispatchEvent(new Event('input'));
    await fixture.whenStable();
    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot();
    expect(component.value).toBe('some text');
  });

  it('should render the highlights correctly based on the matched variables in text', async() => {
    const input = fixture.nativeElement.querySelector('input');
    input.value = 'some text {{variable1}} then {{variable2}} is next';
    input.dispatchEvent(new Event('input'));
    await fixture.whenStable();
    fixture.detectChanges();

    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should render mark element containing the matched variables in text', async() => {
    const input = fixture.nativeElement.querySelector('input');
    input.value = 'some text {{variable1}} then {{variable2}} is next';
    input.dispatchEvent(new Event('input'));
    await fixture.whenStable();
    fixture.detectChanges();

    const marks = fixture.nativeElement.querySelectorAll('mark');

    expect(marks.length).toBe(2);
    expect(marks[0].textContent).toBe('{{variable1}}');
    expect(marks[1].textContent).toBe('{{variable2}}');
  });
});
