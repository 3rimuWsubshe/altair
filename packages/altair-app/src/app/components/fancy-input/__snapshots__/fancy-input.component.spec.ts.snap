// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FancyInputComponent should render the highlights correctly based on the matched variables in text 1`] = `
<div
  id="root2"
>
  <div
    class="fancy-input-container"
  >
    <div
      class="fancy-input-backdrop"
    >
      <div
        class="fancy-input-content fancy-input-highlights"
      >
        
        
        
        <span>
          some text 
        </span>
        
        
        
        
        
        <mark>
          {{variable1}}
        </mark>
        
        
        
        <span>
           then 
        </span>
        
        
        
        
        
        <mark>
          {{variable2}}
        </mark>
        
        
        
        <span>
           is next
        </span>
        
        
      </div>
    </div>
    <input
      class="fancy-input-content fancy-input-element ng-untouched ng-valid ng-dirty"
      placeholder=""
      type="text"
    />
  </div>
</div>
`;

exports[`FancyInputComponent should set component value when input value changes 1`] = `
<div
  id="root1"
>
  <div
    class="fancy-input-container"
  >
    <div
      class="fancy-input-backdrop"
    >
      <div
        class="fancy-input-content fancy-input-highlights"
      >
        
        
        
        <span>
          some text
        </span>
        
        
      </div>
    </div>
    <input
      class="fancy-input-content fancy-input-element ng-untouched ng-valid ng-dirty"
      placeholder=""
      type="text"
    />
  </div>
</div>
`;
